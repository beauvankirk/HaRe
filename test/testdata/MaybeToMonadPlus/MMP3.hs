module MMP3 where

dv :: Int -> Maybe Int -> Maybe Int
dv _ Nothing = Nothing
dv i (Just v) = case i of
  0 -> Nothing
  n -> (Just (v `div` n))
