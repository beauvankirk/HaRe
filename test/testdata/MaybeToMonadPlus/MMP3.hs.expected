module MMP3 where
import Control.Monad

dv :: (MonadPlus m) => Int -> m Int -> m Int
dv i mv = mv >>= (case i of
  0 -> mzero
  n -> (return (v `div` n)))